# Time of sensor data acquisition, and the coordinate frame ID
Header header

# Can message ident0x4D0
uint16 CAN_TX_SCAN_INDEX
uint16 CAN_TX_DSP_TIMESTAMP 
float64 CAN_TX_VEHICLE_SPEED_CALC 
float64 CAN_TX_YAW_RATE_CALC 
float64 CAN_TX_CURVATURE
int8 CAN_TX_LOOK_TYPE 

# Can message ident0x4D1
uint8 CAN_TX_ROLLING_COUNT_2 
string CAN_TX_XCVR_OPERATIONAL 
uint8 CAN_TX_INTERNAL_ERROR 
uint8 CAN_TX_COMM_ERROR 
uint8 CAN_TX_OVERHEAT_ERROR 
string CAN_TX_STEERING_ANGLE 
string CAN_TX_STEERING_ANGLE_SIGN 
int8 CAN_TX_TEMPERATURE 
float64 CAN_TX_YAW_RATE_BIAS 
float64 CAN_TX_VEH_SPD_COMP_FACTOR 
string CAN_TX_ALIGNMENT_STATUS 

# Can message ident0x4D3
float64 CAN_TX_ANGLE_MISALIGNMENT
float64 CAN_TX_AUTO_ALIGN_ANGLE

# Can message ident0x4D4
uint8 DSP_DIAGNOSTIC_BYTE1
uint8 DSP_DIAGNOSTIC_BYTE2
uint8 CAN_BLOCKAGE_BB_BLOCKED
uint8 CAN_BLOCKAGE_MNR_BLOCKED
uint8 CAN_BLOCKAGE_RADAR_BLOCKED
uint8 CAN_TD_BLOCKED 
uint8 CAN_TX_SYSTEM_STATUS 
float64 CAN_TX_YAW_RATE_RAW 
uint8 CAN_TX_YAW_RATE_RAW_QF 
uint8 CAN_TX_YAW_RATE_REF_QF
float64 CAN_TX_YAW_RATE_REFERNCE 

# Can message ident0x600
uint8 CAN_TX_F_VALID_DETECTION_LL_1
uint8 CAN_TX_DETECT_STATUS_LL_1
float64 CAN_TX_DET_AMPLITUTDE_LL_1 
float64 CAN_TX_DET_ANGLE_LL_1
float64 CAN_TX_DET_RANGE_LL_1
float64 CAN_TX_DET_RANGE_RATE_LL_1

# Can message ident0x601
uint8 CAN_TX_F_VALID_DETECTION_ML_1
uint8 CAN_TX_DETECT_STATUS_ML_1
float64 CAN_TX_DET_AMPLITUTDE_ML_1
float64 CAN_TX_DET_ANGLE_ML_1
float64 CAN_TX_DET_RANGE_ML_1
float64 CAN_TX_DET_RANGE_RATE_ML_1 

# Can message ident0x602
uint8 CAN_TX_F_VALID_DETECTION_LL_2
uint8 CAN_TX_DETECT_STATUS_LL_2
float64 CAN_TX_DET_AMPLITUTDE_LL_2
float64 CAN_TX_DET_ANGLE_LL_2
float64 CAN_TX_DET_RANGE_LL_2
float64 CAN_TX_DET_RANGE_RATE_LL_2

# Can message ident0x603
uint8 CAN_TX_F_VALID_DETECTION_ML_2
uint8 CAN_TX_DETECT_STATUS_ML_2
float64 CAN_TX_DET_AMPLITUTDE_ML_2
float64 CAN_TX_DET_ANGLE_ML_2
float64 CAN_TX_DET_RANGE_ML_2
float64 CAN_TX_DET_RANGE_RATE_ML_2

# Can message ident0x604
uint8 CAN_TX_F_VALID_DETECTION_LL_3
uint8 CAN_TX_DETECT_STATUS_LL_3
float64 CAN_TX_DET_AMPLITUTDE_LL_3
float64 CAN_TX_DET_ANGLE_LL_3
float64 CAN_TX_DET_RANGE_LL_3
float64 CAN_TX_DET_RANGE_RATE_LL_3

# Can message ident0x605
uint8 CAN_TX_F_VALID_DETECTION_ML_3
uint8 CAN_TX_DETECT_STATUS_ML_3
float64 CAN_TX_DET_AMPLITUTDE_ML_3
float64 CAN_TX_DET_ANGLE_ML_3
float64 CAN_TX_DET_RANGE_ML_3
float64 CAN_TX_DET_RANGE_RATE_ML_3

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_4
uint8 CAN_TX_DETECT_STATUS_LL_4
float64 CAN_TX_DET_AMPLITUTDE_LL_4
float64 CAN_TX_DET_ANGLE_LL_4
float64 CAN_TX_DET_RANGE_LL_4
float64 CAN_TX_DET_RANGE_RATE_LL_4

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_4
uint8 CAN_TX_DETECT_STATUS_ML_4
float64 CAN_TX_DET_AMPLITUTDE_ML_4
float64 CAN_TX_DET_ANGLE_ML_4
float64 CAN_TX_DET_RANGE_ML_4
float64 CAN_TX_DET_RANGE_RATE_ML_4

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_5
uint8 CAN_TX_DETECT_STATUS_LL_5
float64 CAN_TX_DET_AMPLITUTDE_LL_5
float64 CAN_TX_DET_ANGLE_LL_5
float64 CAN_TX_DET_RANGE_LL_5
float64 CAN_TX_DET_RANGE_RATE_LL_5

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_5
uint8 CAN_TX_DETECT_STATUS_ML_5
float64 CAN_TX_DET_AMPLITUTDE_ML_5
float64 CAN_TX_DET_ANGLE_ML_5
float64 CAN_TX_DET_RANGE_ML_5
float64 CAN_TX_DET_RANGE_RATE_ML_5

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_6
uint8 CAN_TX_DETECT_STATUS_LL_6
float64 CAN_TX_DET_AMPLITUTDE_LL_6
float64 CAN_TX_DET_ANGLE_LL_6
float64 CAN_TX_DET_RANGE_LL_6
float64 CAN_TX_DET_RANGE_RATE_LL_6

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_6
uint8 CAN_TX_DETECT_STATUS_ML_6
float64 CAN_TX_DET_AMPLITUTDE_ML_6
float64 CAN_TX_DET_ANGLE_ML_6
float64 CAN_TX_DET_RANGE_ML_6
float64 CAN_TX_DET_RANGE_RATE_ML_6

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_7
uint8 CAN_TX_DETECT_STATUS_LL_7
float64 CAN_TX_DET_AMPLITUTDE_LL_7
float64 CAN_TX_DET_ANGLE_LL_7
float64 CAN_TX_DET_RANGE_LL_7
float64 CAN_TX_DET_RANGE_RATE_LL_7

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_7
uint8 CAN_TX_DETECT_STATUS_ML_7
float64 CAN_TX_DET_AMPLITUTDE_ML_7
float64 CAN_TX_DET_ANGLE_ML_7
float64 CAN_TX_DET_RANGE_ML_7
float64 CAN_TX_DET_RANGE_RATE_ML_7

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_8
uint8 CAN_TX_DETECT_STATUS_LL_8
float64 CAN_TX_DET_AMPLITUTDE_LL_8
float64 CAN_TX_DET_ANGLE_LL_8
float64 CAN_TX_DET_RANGE_LL_8
float64 CAN_TX_DET_RANGE_RATE_LL_8

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_8
uint8 CAN_TX_DETECT_STATUS_ML_8
float64 CAN_TX_DET_AMPLITUTDE_ML_8
float64 CAN_TX_DET_ANGLE_ML_8
float64 CAN_TX_DET_RANGE_ML_8
float64 CAN_TX_DET_RANGE_RATE_ML_8

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_9
uint8 CAN_TX_DETECT_STATUS_LL_9
float64 CAN_TX_DET_AMPLITUTDE_LL_9
float64 CAN_TX_DET_ANGLE_LL_9
float64 CAN_TX_DET_RANGE_LL_9
float64 CAN_TX_DET_RANGE_RATE_LL_9

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_9
uint8 CAN_TX_DETECT_STATUS_ML_9
float64 CAN_TX_DET_AMPLITUTDE_ML_9
float64 CAN_TX_DET_ANGLE_ML_9
float64 CAN_TX_DET_RANGE_ML_9
float64 CAN_TX_DET_RANGE_RATE_ML_9

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_10
uint8 CAN_TX_DETECT_STATUS_LL_10
float64 CAN_TX_DET_AMPLITUTDE_LL_10
float64 CAN_TX_DET_ANGLE_LL_10
float64 CAN_TX_DET_RANGE_LL_10
float64 CAN_TX_DET_RANGE_RATE_LL_10

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_10
uint8 CAN_TX_DETECT_STATUS_ML_10
float64 CAN_TX_DET_AMPLITUTDE_ML_10
float64 CAN_TX_DET_ANGLE_ML_10
float64 CAN_TX_DET_RANGE_ML_10
float64 CAN_TX_DET_RANGE_RATE_ML_10

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_11
uint8 CAN_TX_DETECT_STATUS_LL_11
float64 CAN_TX_DET_AMPLITUTDE_LL_11
float64 CAN_TX_DET_ANGLE_LL_11
float64 CAN_TX_DET_RANGE_LL_11
float64 CAN_TX_DET_RANGE_RATE_LL_11

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_11
uint8 CAN_TX_DETECT_STATUS_ML_11
float64 CAN_TX_DET_AMPLITUTDE_ML_11
float64 CAN_TX_DET_ANGLE_ML_11
float64 CAN_TX_DET_RANGE_ML_11
float64 CAN_TX_DET_RANGE_RATE_ML_11

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_12
uint8 CAN_TX_DETECT_STATUS_LL_12
float64 CAN_TX_DET_AMPLITUTDE_LL_12
float64 CAN_TX_DET_ANGLE_LL_12
float64 CAN_TX_DET_RANGE_LL_12
float64 CAN_TX_DET_RANGE_RATE_LL_12

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_12
uint8 CAN_TX_DETECT_STATUS_ML_12
float64 CAN_TX_DET_AMPLITUTDE_ML_12
float64 CAN_TX_DET_ANGLE_ML_12
float64 CAN_TX_DET_RANGE_ML_12
float64 CAN_TX_DET_RANGE_RATE_ML_12

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_13
uint8 CAN_TX_DETECT_STATUS_LL_13
float64 CAN_TX_DET_AMPLITUTDE_LL_13
float64 CAN_TX_DET_ANGLE_LL_13
float64 CAN_TX_DET_RANGE_LL_13
float64 CAN_TX_DET_RANGE_RATE_LL_13

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_13
uint8 CAN_TX_DETECT_STATUS_ML_13
float64 CAN_TX_DET_AMPLITUTDE_ML_13
float64 CAN_TX_DET_ANGLE_ML_13
float64 CAN_TX_DET_RANGE_ML_13
float64 CAN_TX_DET_RANGE_RATE_ML_13

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_14
uint8 CAN_TX_DETECT_STATUS_LL_14
float64 CAN_TX_DET_AMPLITUTDE_LL_14
float64 CAN_TX_DET_ANGLE_LL_14
float64 CAN_TX_DET_RANGE_LL_14
float64 CAN_TX_DET_RANGE_RATE_LL_14

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_14
uint8 CAN_TX_DETECT_STATUS_ML_14
float64 CAN_TX_DET_AMPLITUTDE_ML_14
float64 CAN_TX_DET_ANGLE_ML_14
float64 CAN_TX_DET_RANGE_ML_14
float64 CAN_TX_DET_RANGE_RATE_ML_14

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_15
uint8 CAN_TX_DETECT_STATUS_LL_15
float64 CAN_TX_DET_AMPLITUTDE_LL_15
float64 CAN_TX_DET_ANGLE_LL_15
float64 CAN_TX_DET_RANGE_LL_15
float64 CAN_TX_DET_RANGE_RATE_LL_15

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_15
uint8 CAN_TX_DETECT_STATUS_ML_15
float64 CAN_TX_DET_AMPLITUTDE_ML_15
float64 CAN_TX_DET_ANGLE_ML_15
float64 CAN_TX_DET_RANGE_ML_15
float64 CAN_TX_DET_RANGE_RATE_ML_15

# Can message ident0x606
uint8 CAN_TX_F_VALID_DETECTION_LL_16
uint8 CAN_TX_DETECT_STATUS_LL_16
float64 CAN_TX_DET_AMPLITUTDE_LL_16
float64 CAN_TX_DET_ANGLE_LL_16
float64 CAN_TX_DET_RANGE_LL_16
float64 CAN_TX_DET_RANGE_RATE_LL_16

# Can message ident0x607
uint8 CAN_TX_F_VALID_DETECTION_ML_16
uint8 CAN_TX_DETECT_STATUS_ML_16
float64 CAN_TX_DET_AMPLITUTDE_ML_16
float64 CAN_TX_DET_ANGLE_ML_16
float64 CAN_TX_DET_RANGE_ML_16
float64 CAN_TX_DET_RANGE_RATE_ML_16